rfcs/0167-Easy-Taskcluster-setup.md

```diff
@@ -0,0 +1,44 @@
+# RFC 169 - Easy Taskcluster Setup
+* Comments: [#169](https://github.com/taskcluster/taskcluster-rfcs/pull/169)
+* Proposed by: @sarah-clements
+
+# Summary
+
+Make Taskcluster easier to setup into a repository
+
+## Motivation
+
+Currently setting up a Taskcluster config file is confusing compared to its competitors. The [Enhance Taskgraph RFC](https://github.com/mozilla-releng/releng-rfcs/pull/36) will simplify the configuration file needed to get started, but we can go a step further and create a user-friendly experience that will guide the user through setup.
+
+## Assumptions
+
+The solution can be repository host agnostic, but the details of this RFC are based on the Taskcluster integration with Github.
+
+
+# Details
+
+Create a tool or feature that will automatically generate a yaml config file as part of the taskcluster-github app installation and setup. This tool would have presets for common use cases such as running javascript tests or python tests. This might look like a series of questions in a form to help users get started with some default scope, worker type configurations and basic commands.
+
+A second part of this task would be to improve the Github Integration section of the Taskcluster docs to provide details around the installation of the community and firefox-ci apps with github, and clearer guidance around scopes and worker types (for advanced use cases).
+
+## Implementation
+
+How this tool or feature will be implemented is an open question. There are a few different approaches that have been suggested (other idea are also welcome):
+1. After the user installs the taskcluster github app, users will be redirected to a new taskcluster view or other webapp that will walk the user through config setup and commit the config to the repository. 
+
+The advantage of this approach is that it would be in line with what a least one of our competitors offers and would open the door to creating additional value for our users (including where Push Health might fit in). To keep this in scope, additional features - such as what users expect to see or would like to have - can be explored during initial user research and broken down into future milestones.
+
+2. After installing the taskcluster-github app, the user is redirected back to the taskcluster docs. A feature integrated in the UI would walk users through config file setup and generate a yaml file that could be copied into their codebase.
+
+3. The user installs a CLI tool and answers a series of prompts on the command line that then adds the config file to the root directory.
+
```

Comment:
An alternative could be enhancing the existing https://community-tc.services.mozilla.com/quickstart to make it more powerful, and redirect the user to it after they install the taskcluster-github app.

Reply:
This sounds similar to option 1 to me

Reply:
Nice, didn't know about this. I agree building off of and improving this existing tool seems like a great idea.

Reply:
@djmitche Is there a quickstart for firefox-ci too?

Reply:
> This sounds similar to option 1 to me

I think in the beginning option 1 could be limited to just this tool, except it could end up being like the circleCI UI where you also have the pipeline. Or maybe it'd incorporate push health somehow.

Reply:
> An alternative could be enhancing the existing https://community-tc.services.mozilla.com/quickstart to make it more powerful, and redirect the user to it after they install the taskcluster-github app.

I must have missed this in the docs. But yeah, this is pretty much what I was thinking for option 2.

Reply:
Made changes based on this feedback.